pkg/util/lifecycle/lifecycle_utils.go

```diff
@@ -96,6 +98,52 @@ func (c *realControl) UpdatePodLifecycle(pod *v1.Pod, state appspub.LifecycleSta
 	return true, err
 }
 
+func (c *realControl) UpdatePodLifecycleWithHandler(pod *v1.Pod, state appspub.LifecycleStateType, inPlaceUpdateHandler *appspub.LifecycleHook) (bool, error) {
+	if inPlaceUpdateHandler == nil || pod == nil {
```

Comment:
if `inPlaceUpdateHandler` is nil, maybe we should turn to the `UpdatePodLifecycle` instead of directly return?